experimental/Schemes/src/Resolution_structure.jl

```diff
@@ -462,6 +605,31 @@ end
 ##################################################################################################
 # desingularization workers
 ##################################################################################################
+function principalization(I::AbsIdealSheaf; algorithm::Symbol=:BEV)
+  is_one(ideal_sheaf_of_singular_locus(scheme(I))) || error("principalization only available on smooth ambient schemes")
```

Comment:
We do not have is_nonsingular currently, but the question is, whether we need this command. 
in the future: yes
in this PR: no (as this would be off topic and in a completely other file)